### PR TITLE
test: integrate mysql (user-managed) into risedev and inline source tests

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- so that it can be run easier locally.
- also improved tests to include shared source

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: e2e_test/source/cdc_inline/mysql/mysql_create_drop.slt
Comment (xxchan) [inline e2e_test/source/cdc_inline/mysql/mysql_create_drop.slt:1]: This is moved to `e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt`

File: ci/scripts/e2e-source-test.sh [hunk 1]
```diff
@@ -82,15 +82,6 @@ echo "--- cdc share source test"
...
 risedev slt './e2e_test/source/cdc/cdc.share_stream.slt'
 
-# create a share source and check whether heartbeat message is received
-risedev slt './e2e_test/source/cdc/cdc.create_source_job.slt'
-table_id=$(psql -U root -h localhost -p 4566 -d dev -t -c "select id from rw_internal_tables where name like '%mysql_source%';" | xargs);
-table_count=$(psql -U root -h localhost -p 4566 -d dev -t -c "select count(*) from rw_table(${table_id}, public);" | xargs);
-if [ "$table_count" -eq 0 ]; then
-    echo "ERROR: internal table of cdc share source is empty!"
-    exit 1
-fi
```
Comment (xxchan) [inline ci/scripts/e2e-source-test.sh:92]: This is merged into e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt

Comment (xxchan) [reply ci/scripts/e2e-source-test.sh:92]: This test is broken by my pr #16348. That's why I'm trying to make it easier to change..

File: e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+query IT
+select * from tt5;
+----
+5 2023-10-23 14:00:00+00:00
+
+
+query IT
+select * from tt1_shared;
+----
+1 2023-10-23 10:00:00+00:00
```
Comment (xxchan) [inline e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:180]: Hmmm, this test succeeds in CI, but failed for me locally. `tt1` is ok, but `tt1_shared` has a different result. Any ideas?

```
failed to run `e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt`

Caused by:
    query result mismatch:
    [SQL] select * from tt1_shared;
    [Diff] (-expected|+actual)
    -   1 2023-10-23 10:00:00+00:00
    +   1 2023-10-23 02:00:00+00:00
    at e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:177
    
```

Comment (xxchan) [reply e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:180]: cc @xiangjinwu @StrikeW

File: e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt [hunk 3]
```diff
@@ -0,0 +1,218 @@
+# create and drop CDC mysql tables concurrently
+
```
Comment (xxchan) [inline e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:2]: This is the new test file.
Differences with before:
- Now the envvars like `MYSQL_USER` are set by `risedev d` (in the `risedev-env` file), no need to manually `export`. BTW, previous hardcoded `-u root` is unnecessary. 
- added shared source test
- also create the users used by risingwave in the slt.

File: e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt (hunk 2 above)
Comment (xiangjinwu) [reply e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:180]: Most likely a bug in test setup or CDC itself, as the result is different in 8 hours when executed in different time zone.

Comment (StrikeW) [reply e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:180]: In the `MySqlExternalTableReader` which backfill upstream table we have executed `conn.exec_drop("SET time_zone = \"+00:00\"", ()).await?;` before scan the rows. And set `driver.connectionTimeZone=+00:00` in the debezium connector.

File: e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    CREATE USER 'shared-cdc'@'%' IDENTIFIED BY 'abcdef';
+    GRANT SELECT, RELOAD, REPLICATION SLAVE, REPLICATION CLIENT ON *.* TO 'shared-cdc'@'%';
+    #
+    FLUSH PRIVILEGES;
+"
+
+statement ok
+create source s with (
+  ${RISEDEV_MYSQL_WITH_OPTIONS_COMMON},
+  username = 'non-shared-cdc',
```
[the remark this answers is not in the document]
Comment (xxchan) [reply e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:35]:
```suggestion
  username = 'shared-cdc',
```

File: e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt (hunk 2 above)
Comment (xxchan) [reply e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:180]: @StrikeW Sorry that I might not understand the details yet. Do you mean this is a bug or ..?

File: src/risedevtool/src/bin/risedev-dev.rs [hunk 5]
```diff
@@ -348,6 +342,17 @@ fn task_main(
                 ctx.pb
                     .set_message(format!("redis {}:{}", c.address, c.port));
             }
+            ServiceConfig::MySql(c) => {
+                let mut ctx =
+                    ExecuteContext::new(&mut logger, manager.new_progress(), status_dir.clone());
+                // TODO: support starting mysql in RiseDev. Currently it's user-managed only
```
Comment (BugenZhao) [inline src/risedevtool/src/bin/risedev-dev.rs:348]: Shall we check `user_managed` flag here?

File: src/risedevtool/src/bin/risedev-dev.rs [hunk 6]
```diff
@@ -63,6 +62,33 @@ impl ProgressManager {
     }
 }
 
+struct DummyService {
```
Comment (BugenZhao) [inline src/risedevtool/src/bin/risedev-dev.rs:65]: What about moving into `task/dummy_service.rs`?

**Review by BugenZhao — APPROVED** [review]
The RiseDev part LGTM

File: e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt (hunk 2 above)
Comment (StrikeW) [reply e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:180]: I think it should not happen, is it a new test case you added?

Comment (xxchan) [reply e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:180]: Could you try to run this test locally to see whether you can reproduce it?

Comment (StrikeW) [reply e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:180]: Is it a simple way to run the test locally with your branch?

Comment (xxchan) [reply e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:180]: Yes
1. Add your MySQL to risedev user profile
2. risedev d 
3. risedev slt e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt

Comment (xxchan) [reply e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:176]: Let's create a separate issue for this
https://github.com/risingwavelabs/risingwave/issues/16496

**Pull request merged**